### PR TITLE
support azure

### PR DESCRIPTION
**Pull request by PowderLi**
#24944

### CODE REVIEW

**Comment by PowderLi** [comment]
rerun ut

**Comment by PowderLi** [comment]
rerun ut

**Comment by PowderLi** [comment]
rerun ut

File: build/docker/builder/cpu/amazonlinux2023/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ FROM amazonlinux:2023
 
 ARG TARGETARCH
 
-RUN yum install -y wget g++ gcc gdb libstdc++-static git make unzip tar which \
+RUN yum install -y wget g++ gcc gdb libstdc++-static git make zip unzip tar which \
```
Comment (czs007) [inline build/docker/builder/cpu/amazonlinux2023/Dockerfile:16]: After the rebase, this diff should no longer exist.

File: internal/util/indexcgowrapper/build_index_info.go [hunk 2]
```diff
@@ -65,6 +66,7 @@ func NewBuildIndexInfo(config *indexpb.StorageConfig) (*BuildIndexInfo, error) {
 		access_key_value: cAccessValue,
 		root_path:        cRootPath,
 		storage_type:     cStorageType,
+		cloud_provider:   cCloudProvider,
```
Comment (czs007) [inline internal/util/indexcgowrapper/build_index_info.go:69]: Compatibility needs to be taken into consideration here. If the cloudProvider field in the indexpb.StorageConfig passed is empty, it is necessary to fallback to the original logic.

File: internal/proto/index_coord.proto [hunk 3]
```diff
@@ -208,6 +208,7 @@ message StorageConfig {
   string storage_type = 9;
   bool use_virtual_host = 10;
   string region = 11;
+  string cloudProvider = 12;
```
Comment (czs007) [inline internal/proto/index_coord.proto:211]: Compatibility needs to be taken into consideration here.

File: internal/core/src/storage/Types.h [hunk 4]
```diff
@@ -88,6 +88,7 @@ struct StorageConfig {
     std::string access_key_value = "minioadmin";
     std::string root_path = "files";
     std::string storage_type = "minio";
+    std::string cloud_provider = "aws";
```
Comment (czs007) [inline internal/core/src/storage/Types.h:91]: better to be "minio" ?

File: internal/util/indexcgowrapper/build_index_info.go (hunk 2 above)
Comment (czs007) [reply internal/util/indexcgowrapper/build_index_info.go:69]: In the case of it being empty, shall we consider setting the default value as "minio"?

File: internal/core/src/storage/ChunkManager.h [hunk 5]
```diff
@@ -128,6 +128,7 @@ enum ChunkManagerType : int8_t {
     None_CM = 0,
     Local = 1,
     Minio = 2,
+    RemoteStore = 3,
```
Comment (czs007) [inline internal/core/src/storage/ChunkManager.h:131]: just Remote

File: internal/core/src/storage/AzureChunkManager.h [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+#include "storage/Types.h"
+
+#include <string>
+#include <vector>
+
+namespace milvus {
+namespace storage {
+
+/**
+ * @brief This AzureChunkManager is responsible for read and write file in S3.
```
Comment (zhagnlu) [inline internal/core/src/storage/AzureChunkManager.h:32]: not in s3

File: internal/core/src/storage/azure-blob-storage/AzureBlobChunkManager.h [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+#pragma once
+
+#include <stdlib.h>
+#include <iostream>
+#include <azure/storage/blobs.hpp>
+#include "azure/storage/common/storage_exception.hpp"
+
+#include <string>
+#include <vector>
```
Comment (zhagnlu) [inline internal/core/src/storage/azure-blob-storage/AzureBlobChunkManager.h:25]: same as uuper

File: internal/core/src/storage/AzureChunkManager.cpp [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+#include <sstream>
+
+#include "storage/AzureChunkManager.h"
+#include "Exception.h"
+
+namespace milvus {
+namespace storage {
+
+AzureChunkManager::AzureChunkManager(const StorageConfig& storage_config)
+    : default_bucket_name_(storage_config.bucket_name), path_prefix_(storage_config.root_path) {
```
Comment (zhagnlu) [inline internal/core/src/storage/AzureChunkManager.cpp:27]: if not ak mod?  support non-plain text password mode ?

File: internal/core/unittest/test_azure_chunk_manager.cpp [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+using namespace std;
+using namespace milvus;
+using namespace milvus::storage;
+
+StorageConfig
+get_default_storage_config() {
+    auto endpoint = "core.windows.net";
+    auto accessKey = "devstoreaccount1";
+    auto accessValue = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
```
Comment (zhagnlu) [inline internal/core/unittest/test_azure_chunk_manager.cpp:27]: remove it, use a common ak?

File: internal/core/src/storage/Util.cpp [hunk 10]
```diff
@@ -31,7 +33,22 @@
 namespace milvus::storage {
 
 std::map<std::string, ChunkManagerType> ChunkManagerType_Map = {
-    {"local", ChunkManagerType::Local}, {"minio", ChunkManagerType::Minio}};
+    {"local", ChunkManagerType::Local},
+    {"minio", ChunkManagerType::Minio},
+    {"remote", ChunkManagerType::RemoteStore}};
```
Comment (zhagnlu) [inline internal/core/src/storage/Util.cpp:38]: what's mean of remoteStore,  minio is also remote

**Comment by PowderLi** [comment]
rerun ut

File: internal/core/src/storage/Util.cpp (hunk 10 above)
Comment (czs007) [reply internal/core/src/storage/Util.cpp:40]: In the future, configuring it as "remote" will be suggested, with "Minio" being just one of the available providers. @zhagnlu @PowderLi

File: internal/core/src/storage/Util.cpp [hunk 11]
```diff
@@ -31,7 +35,22 @@
...
+
+enum CloudProviderType : int8_t {
+    AWS = 0,
```
Comment (czs007) [inline internal/core/src/storage/Util.cpp:43]: how about add Minio = 0, as the default cloud provider ?

**Comment by PowderLi** [comment]
rerun ut

**Comment by PowderLi** [comment]
rerun ut

**Pull request merged**